packages/cw-orch-core/src/environment/cosmwasm_environment.rs

```diff
@@ -46,6 +46,17 @@ pub trait TxHandler: ChainState + Clone {
         coins: &[cosmwasm_std::Coin],
     ) -> Result<Self::Response, Self::Error>;
 
+    /// Send a Instantiate2Msg to a contract.
+    fn instantiate2<I: Serialize + Debug>(
+        &self,
+        code_id: u64,
+        init_msg: &I,
+        label: Option<&str>,
+        admin: Option<&Addr>,
+        coins: &[cosmwasm_std::Coin],
+        salt: Binary,
+    ) -> Result<Self::Response, Self::Error>;
+
```

Comment:
Is it possible to return (yet) not executed instantiate2 struct instead? Main use for instantiate2 will be knowing the address, before instantiating I'm expecting experience to be something like: 
```rust
let contract1 = chain.instantiate2(...)?;
let contract1_addr = contract1.addr()?;
let contract2 = chain.instantiate2(...)?;
let contract2_addr = contract2.addr()?;
// ...
contract1.submit(Contract1InstantiateMsg{
    contract2_addr: contract2_addr.to_string()
})?;
...
```

Reply:
I'm not sure I understand what you mean here ? You want a function that would return the instantiate2 addr if ever it was instantiated ?

Reply:
> I'm not sure I understand what you mean here ? You want a function that would return the instantiate2 addr if ever it was instantiated ?

Yes, I want function that can return the address of pre-filled, but not executed instantiate2 message.

Reply:
Using helpers such as https://docs.rs/cosmwasm-std/latest/cosmwasm_std/fn.instantiate2_address.html would be annoying, so I would prefer it to be as a part of API

Reply:
OK,implemented what we discussed

Reply:
So this is now a method on the wasm querier correct?